dev-setup.sh

```
@@ -0,0 +1,124 @@
+echo -e "${YELLOW}Starting MongoDB container...${NC}"
+docker-compose up -d db
+
+# Wait for MongoDB to be ready
+echo -e "${YELLOW}Waiting for MongoDB to initialize (this may take a moment)...${NC}"
+for i in {1..30}; do
+    if docker exec todo-mongo mongosh --quiet --eval "db.runCommand({ping:1}).ok" &>/dev/null; then
+        echo -e "\n${GREEN}MongoDB is up and running!${NC}"
+        break
+    fi
+    
+    if [ $i -eq 30 ]; then
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)